### ISSUE TITLE
confirm if world emits all the data we need to reconstruct mud config and ABIs

### ISSUE THREAD

**Issue by holic** [latticexyz/mud#2386]
- [ ] table configs
- [ ] world ABI
- [ ] system ABIs
- [ ] system configs (e.g. access control)

**Comment by holic** [comment]
looks like for generating ABIs, we still need:
- param names (for UX)
- return types
- return type names (for UX)

**Comment by holic** [comment]
I don't think we emit anything for user types or enums, but not sure if that's a blocker for "create MUD config from existing world"?

**Comment by yonadaaa** [comment]
re. https://github.com/latticexyz/mud/issues/2386#issuecomment-1988608215, we discussed this elsewhere and decided the generated ABI will use only the parameters for now instead of emitting all that data on-chain
